src/script/interpreter.cpp

```diff
@@ -1110,35 +1110,46 @@ class CTransactionSignatureSerializer {
 
 } // anon namespace
 
-uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, SigVersion sigversion)
+uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, SigVersion sigversion, CachedHashes* cache)
```

Comment:
Perhaps here we can do the same trick we did with the script/sigcache to keep script/interpreter simpler and more reusable.
ping @sipa